[Regression]: LLVM sanity check may fail due to system `ld` being used

The sanity check introduced in https://github.com/easybuilders/easybuild-easyblocks/pull/3877 tries to build a test program to see if we're missing any dependencies.
While this works in most cases, having a toolchain too old may cause issues because the system linker is used. Neither GCCcore nor LLVM use `binutils` as a dependency (only a `builddependency`), hence trying to invoke the system linker may fail with:

https://github.com/easybuilders/easybuild-easyconfigs/pull/23603#issuecomment-3199592809

```
g++ minimal.cpp -o minimal_cpp $(llvm-config --link-static --system-libs all) failed with exit code 1 (output: /usr/bin/ld.gold: /tmp/boegelbot/ecpr-23603/software/GCCcore/10.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /usr/bin/ld.gold)
```

To fix this, we would either need to move this check to a place where `binutils` is loaded, load `binutils` for this sanity check only, or omit the check entirely...

This is what the issue boils down to:

```
[reuter1@jsczen3l1 ~]$ module purge
[reuter1@jsczen3l1 ~]$ module load GCCcore/10.3.0
[reuter1@jsczen3l1 ~]$ cat test.cpp
int main( void )
{

}
[reuter1@jsczen3l1 ~]$ g++ test.cpp
/usr/bin/ld.gold: /project/def-maintainers/boegelbot/rocky9/zen3/software/GCCcore/10.3.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /usr/bin/ld.gold)
collect2: error: ld returned 1 exit status
[reuter1@jsczen3l1 ~]$ ml binutils/2.36.1-GCCcore-10.3.0
[reuter1@jsczen3l1 ~]$ g++ test.cpp
```

--- a/easybuild/easyblocks/l/llvm.py
+++ b/easybuild/easyblocks/l/llvm.py
@@ -1714,6 +1714,9 @@ class EB_LLVM(CMakeMake):
         # Here, we add the system libraries LLVM expects to find
         minimal_cpp_compiler_cmd += "$(llvm-config --link-static --system-libs all)"
         custom_commands.append(minimal_cpp_compiler_cmd)
+        # binutils is required for the linking step
+        kwargs.setdefault('extra_modules', []).extend(
+            d['short_mod_name'] for d in self.cfg.dependencies() if d['name'] == 'binutils')
 
         return super().sanity_check_step(custom_paths=custom_paths, custom_commands=custom_commands, *args, **kwargs)
 
@@ -1734,7 +1737,6 @@ class EB_LLVM(CMakeMake):
         self.log.debug(f"List of subdirectories for libraries to add to $LD_LIBRARY_PATH + $LIBRARY_PATH: {lib_dirs}")
         self.module_load_environment.LD_LIBRARY_PATH = lib_dirs
         self.module_load_environment.LIBRARY_PATH = lib_dirs
-
         return super().make_module_step(*args, **kwargs)
 
     def make_module_extra(self):

